MWPW-121123 - Aside Block Updates

Sartxi opened the pull request:
- Adjustments to button/link spacing and allowing links in the action areas
- Adds ability to change background color behind text area on Split Asides
- Also contains style updates to the notification aside in order to match consonant designs
- Verified mp4 videos get decorated and laid out properly

Resolves: [MWPW-121123](https://jira.corp.adobe.com/browse/MWPW-121123)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/sarchibeque/asides?martech=off
- After: https://mwpw-121123-aside-block-updates--milo--adobecom.hlx.page/drafts/sarchibeque/asides?martech=off

meganthecoder commented on libs/styles/styles.css:
```diff
@@ -10,222 +10,235 @@
  * governing permissions and limitations under the License.
  */
 
-@import url('variables.css');
-@import url('typography.css');
```
I'm confused about why the line diff for `styles.css` is so large

Sartxi replied:
Yeah not sure how but there were extra spaces in from of nearly everything, fixed. Nice catch 👍🏻

rgclayton reviewed (commented):
Don't merge this yet. I need a little more time to review. 
Adding the do not merge label. I'll remove when finished. 

I should be able to get to this tomorrow. Come yell at me if you need it sooner please.

rgclayton commented on libs/blocks/aside/aside.css:
```diff
@@ -450,93 +469,104 @@
-  .aside.augmented-image .foreground.container .text {
-    flex: 0 0 43%;
+  .aside.right .foreground.container {
+    flex-direction: row-reverse;
```
Silly question. For this new aside right variant. Why wouldn't you just author the text to the right and the image to the left (opposite of default authoring)?

Like this:
![image](https://user-images.githubusercontent.com/2539954/219518794-4c5039cc-d66d-4795-975d-18342c3b08fd.png)

I don't think we need this right variant. Authors should author it that way.

Sartxi replied:
oops this is a reply to the other question, not sure why it shows here @rgclayton

rgclayton replied:
Thanks! That helps.

Sartxi replied:
@rgclayton okay the right alignment is now coming from the authored layout instead of a variant class.

rgclayton commented on libs/blocks/aside/aside.js:
```diff
@@ -14,74 +14,78 @@
+  const html = createTag('video', { playsinline: '', autoplay: '', muted: '', loop: '' }, source);
+  parent.appendChild(html);
+  video.remove();
+}
```
@saugatmalla I'm assuming this won't be needed with [your universal approach](https://github.com/adobecom/milo/pull/478), right? What do you think? 
Can you guys sync up to make sure @Sartxi gets what he needs and we don't have duplicate code?

Nit for future reference: It'd be helpful to see a video in your sample page.


 
cc: @chrischrischris

saugatmalla replied:
> @saugatmalla I'm assuming this won't be needed with [your universal approach](https://github.com/adobecom/milo/pull/478), right? What do you think?

@rgclayton Yes, it won't be needed for the slack uploaded mp4s.

Sartxi replied:
@rgclayton okay I will remove this function. Just as a note the Marquee also uses this method

rgclayton replied:
@Sartxi Is this supposed to be for background videos?

My understanding of maquee's is that .mp4 only worked as a background video.

rgclayton replied:
>@rgclayton Yes, it won't be needed for the slack uploaded mp4s.

I think we need to come to a decisions for .mp4s then to avoid confusion. We shouldn't have to have multiple ways to add .mp4 videos to content. 

cc: @Sartxi

meganthecoder replied:
Some background: The request for the marquee was an inline looped muted video ("animated marquee"), like you see on the [holiday shopping report](https://business.adobe.com/resources/holiday-shopping-report.html). At the time, we didn't have other support for adding those attributes to the video (not sure if there's something with params now).

saugatmalla replied:
Agreed. I believe the [auto decorating of mp4](https://github.com/adobecom/milo/pull/478) can now support mp4s in any block, so we can remove other ways to add .mp4.

Sartxi replied:
I removed the video decorating because that will be included in the other PR from @saugatmalla

rgclayton commented:
>Adds ability to author custom content backgrounds for Split Asides

I don't know what this means. What is custom content backgrounds?

rgclayton commented:
@Sartxi I think we are looking good here minus the video functionality - which @saugatmalla is [working on](https://github.com/adobecom/milo/pull/478).

Do you want to merge this now or wait till he's done with the video player update?

rgclayton commented:
Let's clean up the description to reflect the updates made. Make sure your sample page reflects all changes made to more easily verify everything.

Do we need to worry about any asides that are on live pages having regression issues?

saugatmalla commented on libs/blocks/aside/aside.js:
```diff
@@ -14,74 +14,77 @@
+function decorateVideoContainer(el) {
+  const video = el.querySelector('a[href*=".mp4"]');
+  if (!video) return;
+  video.classList.add('video');
```
It looks like the classes are not being added. The reason could be `.mp4` is already decorated by autoblock.

rgclayton replied:
Is this being left in? Or is the plan to use @saugatmalla autoblock update for videos? 
Is this for background videos? I don't see any video's in your sample page which makes this all a bit confusing...

cc: @ryanmparrish

saugatmalla replied:
@rgclayton Only the classes are being added. I believe it is for the positioning of the video.

Sartxi replied:
@saugatmalla @rgclayton  I made adjustments to avoid needing this function

rgclayton replied:
The decorateVideoContainer function? I still see it as part of your PR.

rgclayton commented:
> * Adds right alignment variant on Standard Asides
> * Adjustments to button/link spacing and allowing links in action areas
> * Adds ability to change background color behind text area on Split Asides
> * Also contains style updates to the notification aside to match consonant designs

@Sartxi The first bullet point is relevant any more, correct? If not, can you clean up the decription of this ticket for historical clarity? 

See my comments above...

Sartxi commented:
> > * Adds right alignment variant on Standard Asides
> > * Adjustments to button/link spacing and allowing links in action areas
> > * Adds ability to change background color behind text area on Split Asides
> > * Also contains style updates to the notification aside to match consonant designs
> 
> @Sartxi The first bullet point isn't relevant any more, correct? If not, can you clean up the decription of this ticket for historical clarity?
> 
> See my comments above...

@rgclayton okay I verified that the auto video decorating is working with these layouts and updated the description. I also added an example to the document.
